crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/class_info.rs

```diff
@@ -275,6 +279,163 @@ mod get_utility_info_tests {
     }
 }
 
+// variants
+// -------
+
+/// The result of matching a variant against a target.
+#[derive(Debug, Eq, PartialEq)]
+enum VariantMatch {
+    /// The variant matches an exact target.
+    Exact,
+    /// The variant matches a partial target.
+    Partial,
+    /// The variant does not match the target.
+    None,
+}
+
+impl VariantMatch {
+    /// Checks if a variant matches a target, and returns the result.
+    fn from(target: &str, variant_text: &str) -> VariantMatch {
+        // If the target matched exactly the variant text.
+        if target == variant_text {
+            return VariantMatch::Exact;
+        };
+
+        // if it has a custom value thus it starts with the variant text and it's followed by "-["
+        if variant_text.starts_with(format!("{}-[", target).as_str()) {
```

Comment:
nit: 

possible optimisation for the future: we should not allocate a new string, use `chars()` check `{}-[` using an iterator.